support/src/main/java/org/jenkinsci/plugins/workflow/support/steps/ExecutorStepExecution.java

```diff
@@ -417,9 +417,12 @@ AccessControlled accessControlled() {
                         cookie = UUID.randomUUID().toString();
                         // Switches the label to a self-label, so if the executable is killed and restarted via ExecutorPickle, it will run on the same node:
                         label = computer.getName();
+
                         EnvVars env = computer.getEnvironment();
                         env.overrideAll(computer.buildEnvironment(listener));
+                        env.put("NODE_NAME", label);
```

Comment:
Incompatible with the core's behavior, because it will return an empty string for Master. "master" must be returned to be compatible with AbstractProjects (see Jenkins code)